File: 1217-relative-sort-array/1217-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        unordered_map<int, int> count;
        for (int num : arr1) {
            count[num]++;
        }

        vector<int> out;
        for (int num : arr2) {
            out.insert(out.end(), count[num], num);
            count.erase(num);
        }

        vector<pair<int, int>> remaining(count.begin(), count.end());
        sort(remaining.begin(), remaining.end());

        for (const auto& [num, freq] : remaining) {
            out.insert(out.end(), freq, num);
        }

        return out;
    }
};

File: 1217-relative-sort-array/1217-relative-sort-array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        count = Counter(arr1)
        arr_2_set = set(arr2)
        counting = []
        for n in arr2:
            counting.extend([n] * count[n])

        temp = []
        for n in arr1:
            if n not in arr_2_set:
                temp.append(n)
        
        temp.sort()
        counting.extend(temp)
        
        return counting
